web/packages/teleport/src/Notifications/Notifications.tsx

```diff
@@ -0,0 +1,359 @@
+/**
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, { useState, useEffect } from 'react';
+import { isBefore } from 'date-fns';
+import styled from 'styled-components';
+import { Alert, Box, Flex, Indicator, Text } from 'design';
+
+import { Notification as NotificationIcon, BellRinging } from 'design/Icon';
+import { useRefClickOutside } from 'shared/hooks/useRefClickOutside';
+import useAttempt from 'shared/hooks/useAttemptNext';
+import { HoverTooltip } from 'shared/components/ToolTip';
+
+import { useTeleport } from 'teleport';
+import useStickyClusterId from 'teleport/useStickyClusterId';
+import { Dropdown } from 'teleport/components/Dropdown';
+
+import { ButtonIconContainer } from 'teleport/TopBar/Shared';
+import { Notification as NotificationType } from 'teleport/services/notifications';
+import { Notification } from './Notification';
+import { useInfiniteScroll } from 'shared/hooks';
+
+const PAGE_SIZE = 15;
+
+export function Notifications({ iconSize = 24 }: { iconSize?: number }) {
+  const ctx = useTeleport();
+  const { clusterId } = useStickyClusterId();
+
+  const { attempt, run } = useAttempt();
+
+  const [userNotificationsStartKey, setUserNotificationsStartKey] =
+    useState('');
+  const [globalNotificationsStartKey, setGlobalNotificationsStartKey] =
+    useState('');
+
+  const [notifications, setNotifications] = useState<NotificationType[]>([]);
+
+  const [userLastSeenNotification, setUserLastSeenNotification] =
+    useState<Date>();
+
+  function fetchNotifications(): Promise<any> {
+    return run(() =>
+      ctx.notificationService
+        .fetchNotifications({
+          clusterId,
+          userNotificationsStartKey: userNotificationsStartKey || undefined,
+          globalNotificationsStartKey: globalNotificationsStartKey || undefined,
+          limit: PAGE_SIZE,
+        })
+        .then(resp => {
+          setUserNotificationsStartKey(resp.userNotificationsNextKey);
+          setGlobalNotificationsStartKey(resp.globalNotificationsNextKey);
```

Comment:
Would this be easier to just store the `resp` instead of different state for each key in the response? the `notifications` state makes sense since we are appending, but the rest seems to just be values from the resp itself. 
maybe something like 

```js
.then(({notifications, ...resp}) => {
```